Kong/kubernetes-ingress-controller
# feat: add KongVault CRD

**What this PR does / why we need it**:


Add definition of `KongVault` CRD to define a custom Kong vault.  

**Which issue this PR fixes**:



fixes #5332, and part of #4559.

**Special notes for your reviewer**:


- How should we validate `backend` and `prefix` in the spec?
  - Currently `backend` only supports several values (env,aws,azure,gcp,hcv) but possible to add new backends. So we only limit it to be non-empty, and let the webhook to call validation API on Kong gateway side.
  - `prefix` should be unique, but I did not find an annotation to tell the APIserver that a field is UNIQUE. So we also have to validate it in webhooks.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR
